core/src/main/java/tc/oc/pgm/flag/Flag.java

```diff
@@ -264,27 +278,48 @@ public boolean canDropAt(Location location) {
   private int sequentialPostCounter = 1;
 
   public Post getReturnPost(Post post) {
+    ImmutableList<Post> posts = definition.getPosts();
+    Post returnPost;
     if (post.isSpecifiedPost()) {
-      return post;
-    }
-    if (predeterminedPost != null) {
-      Post returnPost = predeterminedPost;
+      returnPost = post;
+    } else if (predeterminedPost != null) {
+      returnPost = predeterminedPost;
       predeterminedPost = null;
-      return returnPost;
-    }
-    if (definition.isSequential()) {
-      sequentialPostCounter %= definition.getPosts().size();
-      return definition.getPosts().get(sequentialPostCounter++);
+    } else {
+      ArrayList<Post> possiblePosts = new ArrayList<>(posts);
+      if (definition.isSequential()) {
+        // Start posts collection at sequential count
+        Collections.rotate(possiblePosts, -sequentialPostCounter);
+      } else {
+        Collections.shuffle(possiblePosts);
+      }
+      returnPost = getNextPost(possiblePosts);
     }
-    Random random = match.getRandom();
-    return definition.getPosts().get(random.nextInt(definition.getPosts().size()));
+
+    previousPost = returnPost;
+    return returnPost;
+  }
+
+  private Post getNextPost(Collection<Post> posts) {
+    return posts.stream()
+        .filter(
+            post -> {
+              sequentialPostCounter++;
```

Comment:
This `sequentialPostCounter++;` is in a different location to the PR I made on your repo.

I'm not sure if this still does what you were intending. 
What is the expected functionality of the sequential post counter with posts fail the filter?

Reply:
I moved it here so that the counter skips over posts who's filter returns false. For example, without the counter there, for the posts A, B, and C, if A's respawn filter is always false, the spawning sequence will be BCBBCBBCBB etc. because the post counter doesnt increment extra times when it iterates over a filter returning false.